yt_dlp/extractor/rokfin.py

```diff
@@ -254,3 +422,91 @@ def _real_extract(self, url):
+                if not video_id or not video_type:
+                    self.write_debug(msg=f'skipping {search_result}')
+                    continue
+                yield self.url_result(url=f'{self._BASE_URL}/{video_type}/{video_id}')
+        if not query:
+            return
+        query_data = {'query': query, 'page': {'size': 100}}
+        total_pages = None
+        for page_number in itertools.count(1):
+            query_data['page']['current'] = page_number
+            search_results = self._run_search_query(
+                data=query_data,
+                note='Downloading search results page %d%s' % (page_number, format_field(total_pages, template=' of ~%d') if total_pages and total_pages >= page_number else ''),
+                errnote='Unable to download search results page %d%s' % (page_number, format_field(total_pages, template=' of ~%d') if total_pages and total_pages >= page_number else ''))
+            total_pages = traverse_obj(search_results, ('meta', 'page', 'total_pages'), expected_type=int_or_none)
+            yield from get_video_data(search_results)
+            if not search_results.get('results'):
+                return
+
+    def _run_search_query(self, data, note, errnote):
+        data_bytes = json.dumps(data).encode('utf-8')
+        search_results = self._download_json(
+            self._search_engine_access_url, self._SEARCH_KEY, note=note, errnote=errnote, fatal=False,
+            encoding='utf-8', data=data_bytes, headers={'authorization': self._search_engine_access_key})
+        if search_results is not False:
+            return search_results
+        self.write_debug('updating access credentials')
+        self._search_engine_access_url = self._search_engine_access_key = None
+        self._downloader.cache.store(self._CACHE_SECTION_NAME, 'url', None)
+        self._downloader.cache.store(self._CACHE_SECTION_NAME, 'key', None)
+        self._search_engine_access_url, self._search_engine_access_key = self._get_search_engine_access_credentials()
+        return self._download_json(
+            self._search_engine_access_url, self._SEARCH_KEY, note=note, errnote=errnote, fatal=False,
+            encoding='utf-8', data=data_bytes, headers={'authorization': self._search_engine_access_key}) or {}
+
+    def _get_search_engine_access_credentials(self):
+        notfound_err_page = self._download_webpage('https://rokfin.com/discover', self._SEARCH_KEY, expected_status=404, fatal=False)
+        js_content = ''
+        search_engine_access_url = search_engine_access_key = None
+        for js_file_path in re.finditer(r'<script\s+[^>]*?src\s*=\s*"(?P<path>/static/js/[^">]*)"[^>]*>', notfound_err_page):
+            js_content += self._download_webpage(
+                self._BASE_URL + js_file_path.group('path'), self._SEARCH_KEY,
+                note='Downloading JavaScript file', fatal=False) or ''
```

Comment:
```suggestion
            js_content = self._download_webpage(
                self._BASE_URL + js_file_path.group('path'), video_id,
                note='Downloading JavaScript file', fatal=False) or ''
```
set `video_id` to `None` in real_initialize, but must be provided when running from real_extract

Reply:
The intention behind the `+=` is to concatenate JS files. I assume that a variable assignment (`foo=1`) may **theoretically** be split over multiple files.

Reply:
Why do you need to concatenate since you already search the previous files?

Reply:
Because the string being matched may **theoretically** be broken up into multiple pieces, each in its own JS file. The files must be put together for the search to succeed.

Reply:
how can a dict be split b/w 2 js files? 😕

Reply:
Why not (in theory)? Does the JS standard prohibit this?

Reply:
In theory JS files can be split, but never at the middle of any object (dict in Python).

Reply:
Thanks a lot, @Lesmiscore. By any chance, can you send a link?

Reply:
You can't obviously do that like Python can't! https://replit.com/@Lesmiscore/test-js

Reply:
@pukkandan 

> set video_id to None in real_initialize, but must be provided when running from real_extract

Can you clarify?

Reply:
When using `_download_webpage` etc from `_real_initialize`, pass `video_id=None`

But from `_real_extract`, it should be an id of the video/playlist (Not necessarily the final `video_id`, just something that can help identify it)